Src/IronPython.Modules/_struct.cs

```diff
@@ -538,6 +547,72 @@ public class Struct : IWeakReferenceable {
             #endregion
         }
 
+        [PythonType("unpack_iterator"), Documentation("Represents an iterator returned by _struct.iter_unpack()")]
+        public class PythonUnpackIterator : System.Collections.IEnumerator, System.Collections.IEnumerable {
+            private object _iter_current;
+            private int _next_offset;
+
+            private CodeContext _context;
+            private IList<byte> _buffer;
+            private int _start_offset;
+            private Struct _owner;
+
+            private PythonUnpackIterator() { }
+
+            internal PythonUnpackIterator(Struct/*!*/ owner, CodeContext/*!*/ context, IList<byte>/*!*/ buffer, int offset) {
+                _context = context;
+                _buffer = buffer;
+                _start_offset = offset;
+                _owner = owner;
+
+                Reset();
+            }
+
+            internal PythonUnpackIterator(Struct/*!*/ owner, CodeContext/*!*/ context, ArrayModule.array/*!*/ buffer, int offset) {
+                _context = context;
+                _buffer = buffer.ToByteArray();
+                _start_offset = offset;
+                _owner = owner;
+
+                Reset();
+            }
+
+            public System.Collections.IEnumerator GetEnumerator() {
+                return this;
+            }
+
+            public object Current => _iter_current;
```

Comment:
Can you add a `PythonHidden` attribute here so the .NET members don't appear in `dir`? Similaly for `GetEnumerator`, `MoveNext` and `Reset`.
```
>>> dir(struct.iter_unpack("h", b""))
['Current', 'GetEnumerator', 'MoveNext', 'Reset', '__class__', ...
```

```C#
[PythonHidden]
public object Current => _iter_current;
```

Reply:
Added the attribute in 8830349.